handlers/handlers.go

```diff
@@ -46,74 +55,284 @@ func parseReqBody(w http.ResponseWriter, body io.ReadCloser, bodyObj *models.Pre
 	return nil
 }
 
+func parseStringToInt(strings ...string) ([]int, error) {
+	var (
+		val int
+		err error
+	)
+
+	vals := make([]int, 0)
+	for _, str := range strings {
+		val = 0
+		if str != "" {
+			val, err = strconv.Atoi(str)
+			if err != nil {
+				errMsg := "failed to convert string to int: " + err.Error()
+				return nil, errors.New(errMsg)
+			}
+		}
+		vals = append(vals, val)
+	}
+
+	return vals, nil
+}
+
 // PostPrefsHandler creates new preferences for a user
-func PostPrefsHandler(w http.ResponseWriter, r *http.Request) {
+func (env *Env) PostPrefsHandler(w http.ResponseWriter, r *http.Request) {
 	defer r.Body.Close()
 
 	reqBody := models.NewPreferences()
 	if err := parseReqBody(w, r.Body, reqBody); err != nil {
 		return
 	}
 
+	vals, err := parseStringToInt(r.Header.Get("User-ID"))
+	if err != nil {
+		log.Println(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	reqBody.UserID = vals[0]
+
+	if err = env.DB.CreatePrefs(reqBody); err != nil {
+		errMsg := fmt.Sprintf(
+			"failed to create prefs (%+v): %s",
+			*reqBody,
+			err.Error(),
+		)
+		log.Println(errMsg)
+		responseCode := http.StatusInternalServerError
+		if err == models.ErrPrefsExists {
+			responseCode = http.StatusConflict
+		}
+		http.Error(w, errMsg, responseCode)
+		return
+	}
+
+	reqBody.UserID = 0
+
+	w.Header().Set("Content-Type", "application/json")
 	json.NewEncoder(w).Encode(reqBody)
```

Comment:
This should maybe return `201 Created` instead of `200 OK`.

Reply:
Yeah I was going between `201 Created` and `200 OK` because they are some articles that I saw that said that `201 Created` should always return a location for the created resource but since the resources don't really "exist" anywhere (like reachable by URL) I just went with `200 OK`.

Reply:
[This page](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201) says "...the new resource is returned in the body of the message, its location being either **the URL of the request**, or the content of the `Location` header", so maybe it's okay to use `200 OK`? Or did another article contradict that point?

Reply:
Okay so I assumed "its location being either the URL of the request, or the content of the Location header" means that the `Location` header of the response should be set to either of those two options and [this specification](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.30) says that the `Location` header is used to redirect the user to the new resource so I was worried that the browser would automatically try to redirect to that, but honestly I didn't really look into that worry lol

Reply:
I just caught that I said `200 OK` instead of `201 Created` in my last comment which might've been confusing, so I fixed that.

Reply:
Lol yeah I forgot to correct you but I understood.

Reply:
Anyway, it seems like doing `201 Created` without the `Location` header is a legit thing, from what I'm reading. Or do you think sticking with `200 OK` is better?

Reply:
Nah I'm definitely good with switching to `201 Created`. I just wanted to give the reasons for why I didn't go with it initially.